web/client/components/map/leaflet/Feature.jsx

```diff
@@ -158,4 +160,25 @@ class Feature extends React.Component {
     }
 }
 
+// the _msLegacyGeoJSON flag has been added in case a vector layer is using the feature component to manage style and click action
+// in case of this key is missing the layer will be in charge to manage the styling content of the features
+// layers that are still using _msLegacyGeoJSON are annotations, highlights or vector layer with default legacy style
+class Feature extends React.Component {
+    static propTypes = {
+        container: PropTypes.object
+    }
+    render() {
+        return this.props.container._msLegacyGeoJSON
+            ? <FeatureComponent
```

Comment:
i've missed how this are rendered now with new styiling system since you no longer render a component